beacon_node/execution_layer/src/engine_api/http.rs

```diff
@@ -0,0 +1,1061 @@
+    fn generate_transactions<E: EthSpec>(
+        spec: &[usize],
+    ) -> VariableList<Transaction<E::MaxBytesPerTransaction>, E::MaxTransactionsPerPayload> {
+        let mut txs = VariableList::default();
+
+        for &num_bytes in spec {
+            let mut tx = VariableList::default();
+            for _ in 0..num_bytes {
+                tx.push(0).unwrap();
+            }
+            txs.push(tx).unwrap();
+        }
+
+        txs
+    }
+
+    #[test]
+    fn transaction_serde() {
+        assert_transactions_serde::<MainnetEthSpec>(
+            "empty",
+            generate_transactions::<MainnetEthSpec>(&[]),
+            json!([]),
+        );
+        assert_transactions_serde::<MainnetEthSpec>(
+            "one empty tx",
+            generate_transactions::<MainnetEthSpec>(&[0]),
+            json!(["0x"]),
+        );
+        assert_transactions_serde::<MainnetEthSpec>(
+            "two empty txs",
+            generate_transactions::<MainnetEthSpec>(&[0, 0]),
+            json!(["0x", "0x"]),
+        );
+        assert_transactions_serde::<MainnetEthSpec>(
+            "one one-byte tx",
+            generate_transactions::<MainnetEthSpec>(&[1]),
+            json!(["0x00"]),
+        );
+        assert_transactions_serde::<MainnetEthSpec>(
+            "two one-byte txs",
+            generate_transactions::<MainnetEthSpec>(&[1, 1]),
+            json!(["0x00", "0x00"]),
+        );
+        assert_transactions_serde::<MainnetEthSpec>(
+            "mixed bag",
+            generate_transactions::<MainnetEthSpec>(&[0, 1, 3, 0]),
+            json!(["0x", "0x00", "0x000000", "0x"]),
+        );
+
+        /*
+         * Check for too many transactions
+         */
+
+        let num_max_txs = <MainnetEthSpec as EthSpec>::MaxTransactionsPerPayload::to_usize();
+        let max_txs = (0..num_max_txs).map(|_| "0x00").collect::<Vec<_>>();
+        let too_many_txs = (0..=num_max_txs).map(|_| "0x00").collect::<Vec<_>>();
+
+        decode_transactions::<MainnetEthSpec>(serde_json::to_value(max_txs).unwrap()).unwrap();
+        assert!(
+            decode_transactions::<MainnetEthSpec>(serde_json::to_value(too_many_txs).unwrap())
+                .is_err()
+        );
+
+        /*
+         * Check for transaction too large
+         */
+
+        use eth2_serde_utils::hex;
+
+        let num_max_bytes = <MainnetEthSpec as EthSpec>::MaxBytesPerTransaction::to_usize();
+        let max_bytes = (0..num_max_bytes).map(|_| 0_u8).collect::<Vec<_>>();
+        let too_many_bytes = (0..=num_max_bytes).map(|_| 0_u8).collect::<Vec<_>>();
+        decode_transactions::<MainnetEthSpec>(
+            serde_json::to_value(&[hex::encode(&max_bytes)]).unwrap(),
+        )
+        .unwrap();
+        assert!(decode_transactions::<MainnetEthSpec>(
+            serde_json::to_value(&[hex::encode(&too_many_bytes)]).unwrap()
+        )
+        .is_err());
```

Comment:
This test is the cause of the CI failure on Windows. It allocates 2GB of memory (as 1GB + 1GB) due to `MaxBytesPerTransaction` now being equal to 1GB.

The `STATUS_STACK_BUFFER_OVERRUN` error code identified by Paul [here](https://github.com/sigp/lighthouse/pull/2781#issuecomment-964787275) is indicative of a "fast abort" on Windows, and indeed that stack trace also includes a line: `memory allocation of 2147483650 bytes failed`.

We could try testing the two 1GB transactions separately (with `drop` in between), but even then 1GB is still a lot to allocate and it might still fail. I'll raise a PR to try it I guess...

More on the `STATUS_STACK_BUFFER_OVERRUN`: https://devblogs.microsoft.com/oldnewthing/20190108-00/?p=100655

Reply:
I think the lowest we can go is 3GB allocated, because the hex encoding doubles the size of the transaction (1GB bytes + 2GB hex bytes). The CI machines allegedly have 7GB of RAM so this might be OK: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

Reply:
Can probably do 2GB with some string hacks. Trying that now

Reply:
Resolved via #2809 :pray: